### ISSUE TITLE
Wildcard items issue

### ISSUE THREAD

**Issue by sbouwmeester** [sergeyshushlyapin/Sitecore.FakeDb#163]
Hi,

I am working on website generator solution. I am creating a website folder where we want to implement a newsoverview page with a newsdetail page based on the wildcard solution. The name of the wildcard page is always a '*'. In my project this feature is allowed, but in my unit tested version of this code this isn't allowed. I get the following error:
"Sitecore.Exceptions.InvalidItemNameException : An item name must satisfy the pattern: ^\w[\w\s]*$ (controlled by the setting ItemNameValidation)"

You can check this behaviour by implementing the following unit test:

 [Fact]
        public void ShouldAddWildcard()
        {
            using (var db = new Db())
            {
                var item = db.GetItem("/sitecore/content/");
                item.Add("*", new TemplateID(Sitecore.TemplateIDs.Folder));
            }
        }

Am I doing something wrong, or is the wildcard name not supported by fakedb?

Greetings,
Sander

**Comment by sergeyshushlyapin** [comment]
Interesting catch, thanks @sbouwmeester!

There is the ItemNameValidation setting in the _default_ `Sitecore.config` file. If you add it to your tests project `App.config` the test will pass:
```
<setting name="ItemNameValidation" value="^[\w\*\$][\w\s\-\$]*(\(\d{1,}\)){0,1}$" />
```

This setting is missing in FakeDb configs, so the default value specified in the `Sitecore.Configuration.Settings.ItemNameValidation` property is taken (`@"^\w[\w\s]*$`). I'll add the setting to FakeDb configuration so that it behaves more like the default item validation.

**Comment by sbouwmeester** [comment]
Thanks! I will add this entry! 

Greetings,
Sander

**Comment by sergeyshushlyapin** [comment]
Released as [v1.4.1](https://www.nuget.org/packages/Sitecore.FakeDb/).
